src/client/cl_scrn.c

```diff
@@ -117,22 +117,66 @@ void SCR_DrawPic(float x, float y, float width, float height, qhandle_t hShader)
  */
 static void SRC_DrawSingleChar(int x, int y, int w, int h, int ch)
 {
-	int   row, col;
-	float frow, fcol;
-	float size;
+	if (!cls.consoleFont.glyphScale || (!cls.etIconFont.glyphScale && ch < 32))
+	{
+		int   row, col;
+		float frow, fcol;
+		float size;
+
+		ch &= 255;
+
+		row = ch >> 4;
+		col = ch & 15;
 
-	ch &= 255;
+		frow = row * 0.0625f;
+		fcol = col * 0.0625f;
+		size = 0.0625f;
 
-	row = ch >> 4;
-	col = ch & 15;
+		re.DrawStretchPic(x, y, w, h, fcol, frow,
+		                  fcol + size, frow + size,
+		                  cls.charSetShader);
+	}
+	else
+	{
+		float       scaleX, scaleY, scale;
+		glyphInfo_t *info;
 
-	frow = row * 0.0625f;
-	fcol = col * 0.0625f;
-	size = 0.0625f;
+		if (ch < 32)
+		{
+			info = Q_UTF8_GetGlyphVanilla(&cls.etIconFont, ch);
+		}
+		else
+		{
+			info = Q_UTF8_GetGlyphExtended(&cls.consoleFont, ch);
+		}
 
-	re.DrawStretchPic(x, y, w, h, fcol, frow,
-	                  fcol + size, frow + size,
-	                  cls.charSetShader);
+		if (!info)
+		{
+			return;
+		}
+
+		scale = cls.consoleFont.glyphScale;
```

Comment:
Scaling still seems to be a bit off. Default size is bigger than before now. But also on different resolutions. Is that what the FIXME is about?

Reply:
Yes, the scaling is not 1 to 1 matching, and It can never really be so. The original font is squashed and mangled to fit a 16x16 image grid. Hard to match that perfectly and don't know if really needed.
This PR is meant as the final, ultimate and best solution.

Reply:
Unless we already have a way to control the font size/points that I am not aware of a CVAR for it would be helpful in the ultimate solution 👍 Besides that, is there a way to reload the font for users, besides that `cl_font` that was meant for debugging?